src/analytics/redux/tracked_events.js

```diff
@@ -19,6 +69,41 @@ function getBlockObject( block ) {
 	return select( 'core/block-editor' ).getBlock( block ) || {};
 }
 
+/**
+ * Track block replacement.
+ *
+ * @param {Array}          originalBlockIds ids or blocks that are being replaced
+ * @param {Object | Array} blocks           block instance object or an array of such objects
+ * @param {Array}          args             additional insertBlocks data e.g. metadata containing inserter method.
+ * @return {void}
+ */
+const trackBlockReplacement = ( originalBlockIds, blocks, ...args ) => {
+	const metaData = args?.[ 3 ] ?? {};
+	const insert_method = getBlockInserterUsed( originalBlockIds, metaData );
+
+	trackBlocksHandler( blocks, 'editor_block_inserted', ( { name } ) => ( {
+		block_name: name,
+		insert_method,
+	} ) );
+};
+
+/**
+ * Track block insertion.
+ *
+ * @param {Object | Array} blocks block instance object or an array of such objects
+ * @param {Array}          args   additional insertBlocks data e.g. metadata containing inserter method.
+ * @return {void}
+ */
+function trackBlockInsertion( blocks, ...args ) {
+	const metaData = args?.[ 3 ] ?? {};
```

Comment:
I'm wondering if, instead of accessing the arguments by index, we could directly fetch the `meta` argument defined in the arguments of the `insertBlock` Redux action:

https://github.com/WordPress/gutenberg/blob/d27c43fda419995a80b1cbdeafe2b24d9a0e2164/packages/block-editor/src/store/actions.js#L534-L540

I.e. apply the following change:
```
trackBlockInsertion( blocks, ...args ) {
```
=>
```
function trackBlockInsertion(
	blocks,
	index,
	rootClientId,
	updateSelection,
	initialPosition,
	meta
) {
```

Reply:
Updated it in https://github.com/wordpress-mobile/gutenberg-mobile/pull/5976/commits/7277128f7526d0557303f58a89bca48120c3f88e and https://github.com/wordpress-mobile/gutenberg-mobile/pull/5976/commits/effe376cb59ea829723d0d8538e4bfc8731441db#diff-b264ddd07fecd2f9655557832fb8f89baa62f7f89c387e80df664c26ceb36084